src/test/java/com/salesforce/storm/spout/dynamic/kafka/ConsumerTest.java

```diff
@@ -272,10 +275,14 @@ public void testTimedFlushConsumerState() throws InterruptedException {
         );
         when(mockKafkaConsumer.partitionsFor(eq(topicName))).thenReturn(mockPartitionInfos);
 
+        final ConsumerRecords<byte[], byte[]> consumerRecords = makeConsumerRecords(0, 1);
+        when(mockKafkaConsumer.poll(300)).thenReturn(consumerRecords);
+
         // Call constructor
         final Consumer consumer = new Consumer(mockKafkaConsumer);
         consumer.setClock(mockClock);
         consumer.open(config, getDefaultVSpoutId(), getDefaultConsumerCohortDefinition(), mockPersistenceAdapter, null);
+        consumer.nextRecord();
```

Comment:
@Crim By dropping the position() call, the first call to nextRecord() gets the state in the right spot on a reset to the beginning. This seems ok to me, yeah? Do we really need to be persisting state before we've done a poll()?